zerver/migrations/0191_customprofilefieldvalue_rendered_value.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.16 on 2018-11-14 12:15
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('zerver', '0190_cleanup_pushdevicetoken'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='customprofilefieldvalue',
+            name='rendered_value',
+            field=models.TextField(default=None, null=True),
+        ),
```

Comment:
So, I think we need to write a database migration to update all historical CustomProfileFieldValue objects to have the right format.  I think the code needs to look something like this:
```
for field_value in CustomProfileFieldValue.objects.select_related("field").all()
    if field_value.field.field_type in [CustomProfileField.SHORT_TEXT, CustomProfileField.LONG_TEXT]:
        field_value.rendered_value = bugdown_convert(field_value.value, message_realm=field_value.field.realm)
```

(But see https://zulip.readthedocs.io/en/latest/subsystems/schema-migrations.html for more docs on this)

That way, we don't need to worry about the database having a mix of cases with `rendered_value` and others without it.

Reply:
Done :slightly_smiling_face:.

Reply:
I'm faced with the issue of importing the bugdown conversion function in the migrations file that I modified. I read the suggested docs and feel like maybe I might need to use a "#nolint" comment here (it feels unavoidable). Any advice?

Reply:
@timabbott, I'm unsure of how I could call the `bugdown_convert()` function inside of the custom migrations code. I've gone through the document that you've suggested, and I realize that importing that code is not a possibility here. Nor could I copy and paste that code. So could you give me some advice on the best thing to do would be in this case?

Reply:
Commented on this in PMs, but I think you're right that we don't want to do this via a migration.  So, instead, we should make the code for fetching one of these field types with a `rendered_content` of `None` automatically fix the field (by rendering the content) and save it back to the database.  And we'll have to keep that backwards-compatibility code around for a few years, which is not the end of the world.

Reply:
@timabbott, so I'm adding this check in the `get_raw_user_data()` method which is ultimately used in `fetch_initial_state_data()`. Now what I noticed is that while I've added code for sending rendered_value on updates to the field value, I wasn't sending it in the initial state. So I'm trying to add that now. However, this will mean that I will first need to better familiarize myself with the code in _events.py_.

Reply:
This will be done in a subsequent PR where I would also have to add sending the rendered_value in the initial state data (since this would also require some frontend changes).